src/manage_nsfs/nc_lifecycle.js

```diff
@@ -518,32 +544,89 @@ class NCLifecycle {
                 filtered_objects.push(candidate);
             }
         });
-
-        const bucket_state = this.lifecycle_run_status.buckets_statuses[bucket_json.name].state;
-        bucket_state.num_processed_objects += objects_list.objects.length;
-        if (objects_list.is_truncated) {
-            rule_state.key_marker = objects_list.next_marker;
-        } else {
-            rule_state.is_finished = true;
-        }
         return filtered_objects;
     }
 
+    filter_expired_delete_marker(object, next_object, filter_func) {
+        const lifecycle_info = this._get_lifecycle_object_info_from_list_object_entry(object);
+        if (!filter_func(lifecycle_info)) return false;
+        return object.is_latest && object.delete_marker && object.key !== next_object.key;
+    }
+
     /**
      * get_candidates_by_expiration_delete_marker_rule processes the expiration delete marker rule
      * @param {*} lifecycle_rule
      * @param {Object} bucket_json
      * @returns {Promise<Object[]>}
      */
-    async get_candidates_by_expiration_delete_marker_rule(lifecycle_rule, bucket_json) {
-        // TODO - implement
-        return [];
+    async get_candidates_by_expiration_delete_marker_rule(lifecycle_rule, bucket_json, object_sdk, {versions_list}) {
+        const rule_state = this.lifecycle_run_status.buckets_statuses[bucket_json.name].rules_statuses[lifecycle_rule.id].state.noncurrent;
+        if (rule_state.is_finished) return [];
+        if (!versions_list) {
+            versions_list = await this.load_versions_list(object_sdk, bucket_json, rule_state);
+        }
+        const candidates = [];
+        const filter_func = this._build_lifecycle_filter({filter: lifecycle_rule.filter, expiration: 0});
```

Comment:
when having an expired delete marker we don't care if there is also expiration by days/date?

Reply:
its a separate rule than both of them

Reply:
according to aws:
```
You can't specify both a Days and an ExpiredObjectDeleteMarker tag on the same rule. When you specify the Days tag, Amazon S3 automatically performs ExpiredObjectDeleteMarker cleanup when the delete markers are old enough to satisfy the age criteria. To clean up delete markers as soon as they become the only version, create a separate rule with only the ExpiredObjectDeleteMarker tag.
```
so I guess you are correct. I'll change this

Reply:
trying to put date and expire delete marker together returns MalformedXML error. there is nothing on AWS documentation about about implicit ExpiredObjectDeleteMarker when having date rule. so I guess we can't have both of them together